ground/src/test/java/com/google/android/ground/persistence/sync/SurveySyncWorkerTest.kt

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.ground.persistence.sync
+
+import android.content.Context
+import android.content.res.Resources.NotFoundException
+import androidx.test.core.app.ApplicationProvider
+import androidx.work.ListenableWorker.Result
+import androidx.work.WorkerFactory
+import androidx.work.WorkerParameters
+import androidx.work.testing.SynchronousExecutor
+import androidx.work.testing.TestWorkerBuilder
+import androidx.work.workDataOf
+import com.google.android.ground.BaseHiltTest
+import com.google.android.ground.domain.usecases.survey.SyncSurveyUseCase
+import com.google.android.ground.persistence.sync.SurveySyncWorker.Companion.SURVEY_ID_PARAM_KEY
+import com.google.common.truth.Truth.assertThat
+import com.sharedtest.FakeData.SURVEY
+import dagger.hilt.android.testing.BindValue
+import dagger.hilt.android.testing.HiltAndroidTest
+import java.util.concurrent.Executor
+import java.util.concurrent.Executors
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.Mockito.`when`
+import org.robolectric.RobolectricTestRunner
+
+@HiltAndroidTest
+@OptIn(ExperimentalCoroutinesApi::class)
+@RunWith(RobolectricTestRunner::class)
+class SurveySyncWorkerTest : BaseHiltTest() {
+  private lateinit var context: Context
+  private lateinit var executor: Executor
+  @BindValue @Mock lateinit var syncSurvey: SyncSurveyUseCase
+
+  private val factory =
+    object : WorkerFactory() {
+      override fun createWorker(
+        appContext: Context,
+        workerClassName: String,
+        workerParameters: WorkerParameters
+      ) = SurveySyncWorker(appContext, workerParameters, syncSurvey)
+    }
+
+  @Before
+  override fun setUp() {
+    super.setUp()
+    context = ApplicationProvider.getApplicationContext()
+    executor = Executors.newSingleThreadExecutor()
+  }
+
+  @Test
+  fun doWork_FailsOnNoInput() {
+    val worker =
+      TestWorkerBuilder<SurveySyncWorker>(
+          context,
+          executor,
+          inputData = workDataOf(Pair(SURVEY_ID_PARAM_KEY, null)),
+        )
+        .setWorkerFactory(factory)
+        .build()
+    val result = worker.doWork()
```

Comment:
Can we create an abstract class for creating the worker along with the dependencies and calling `.doWork()`? 

Something like:

```
fun createWorkerAndDoWork<T>(args: Pair) {}
```

Reply:
If we wind up needing more of these that sounds like a good idea; for now I don't think the extra indirection is worth it yet.